odoo/addons/base/models/ir_model.py

```diff
@@ -1765,28 +1765,17 @@ def check(self, model, mode='read', raise_exception=True):
 
         self.flush(self._fields)
 
-        # We check if a specific rule exists
+        # We check if a specific or generic rule exists
         self._cr.execute("""SELECT MAX(CASE WHEN perm_{mode} THEN 1 ELSE 0 END)
                               FROM ir_model_access a
                               JOIN ir_model m ON (m.id = a.model_id)
-                              JOIN res_groups_users_rel gu ON (gu.gid = a.group_id)
+                              LEFT JOIN res_groups_users_rel gu ON (gu.gid = a.group_id)
```

Comment:
My two cents:
```suggestion
                         LEFT JOIN res_groups_users_rel gu ON (gu.gid = a.group_id)
```

Couldn't be using `bool_or` in the select clause better for performances too? I would assume that the `max` needs to check all the rows, but the `bool_or` can stop quicker to do the aggregation

Reply:
FYI it was the base back then, but was changed to be compatible with postgresql 7.4
https://github.com/odoo/odoo/commit/8579038ac7029163c9c631a7f6fec14f7b712637

According to the lastest docs (are they up to date?) we are only supporting 10.0 and up
https://www.odoo.com/documentation/15.0/administration/install/install.html#postgresql